mono/metadata/icall.c

```diff
@@ -6752,7 +6752,7 @@ ves_icall_System_Environment_get_UserName (MonoError *error)
 {
 	error_init (error);
 	/* using glib is more portable */
-	return mono_string_new_handle (mono_domain_get (), g_get_user_name (), error);
+	return g_get_user_name () != NULL ? mono_string_new_handle (mono_domain_get (), g_get_user_name (), error) : MONO_HANDLE_CAST (MonoString, NULL_HANDLE);
```

Comment:
Comparison to null is optional, and adds operator precedence ambiguity.

Reply:
There is also a shorthand for MONO_HANLE_CAST (MonoString, NULL_HANDLE) available, but that is only cosmetic, ok either way.

Reply:
OK.